oidc-ui/src/components/Form.js

```diff
@@ -0,0 +1,261 @@
+      if (errors != null && errors.length > 0) {
+        setError({
+          prefix: "authentication_failed_msg",
+          errorCode: errors[0].errorCode,
+          defaultMsg: errors[0].errorMessage,
+        });
+        return;
+      } else {
+        setError(null);
+
+        let nonce = openIDConnectService.getNonce();
+        let state = openIDConnectService.getState();
+
+        let params = buildRedirectParams(
+          nonce,
+          state,
+          openIDConnectService.getOAuthDetails(),
+          response.consentAction
+        );
+
+        navigate(process.env.PUBLIC_URL + "/consent" + params, {
+          replace: true,
+        });
+      }
+    } catch (error) {
+      setError({
+        prefix: "authentication_failed_msg",
+        errorCode: error.message,
+        defaultMsg: error.message,
+      });
+      setStatus(states.ERROR);
+    }
+  };
+
+  useEffect(() => {
+    let loadComponent = async () => {
+      i18n.on("languageChanged", () => {
+        if (showCaptcha) {
+          //to rerender recaptcha widget on language change
+          setShowCaptcha(false);
+          setTimeout(() => {
+            setShowCaptcha(true);
+          }, 1);
+        }
+      });
+    };
+
+    loadComponent();
+  }, []);
+
+  useEffect(() => {
+    setInvalidState(!Object.values(loginState).every((value) => value?.trim()));
+  }, [loginState]);
+
+  const onCloseHandle = () => {
+    let tempBanner = errorBanner.map((_) => _);
+    tempBanner[0].show = false;
+    setErrorBanner(tempBanner);
+  };
+
+  return (
+    <>
+      <div className="grid grid-cols-8 items-center">
+        <div className="h-6 items-center text-center flex items-start">
+          <button
+            onClick={() => handleBackButtonClick()}
+            className="text-sky-600 text-2xl font-semibold justify-left rtl:rotate-180"
+          >
+            &#8592;
+          </button>
+        </div>
+        <div className="h-6 flex justify-center col-start-2 col-span-6 h-fit">
+          <h1
+            className="text-center text-sky-600 font-semibold line-clamp-2"
+            title={t("sign_in_with_details")}
+          >
+            {t("sign_in_with_details")}
+          </h1>
+        </div>
+      </div>
+
+      {errorBanner.length > 0 && (
+        <ErrorBanner
+          showBanner={errorBanner[0]?.show}
+          errorCode={errorBanner[0]?.errorCode}
+          onCloseHandle={onCloseHandle}
+        />
+      )}
+
+      <form className="mt-6 space-y-6" onSubmit={handleSubmit}>
+        {fields.map((field) => (
+          <div className="-space-y-px">
+            <InputWithImage
+              key={"Form_" + field.id}
+              handleChange={handleChange}
+              value={loginState["Form_" + field.id]}
+              labelText={t(field.id)}
+              labelFor={field.id}
+              id={"Form_" + field.id}
+              type={field.type}
+              isRequired={true}
+              placeholder={t(field.id)}
```

Comment:
Placeholder and labeltext  are same? whats the point? can we not have field.placeholder?

Reply:
There is no separate placeholder label to display used the same id as placeholder